src/allocator_defrag.c

```diff
@@ -0,0 +1,484 @@
+    je_mallctl("epoch", &epoch, &sz, &epoch, sz);
+    char buf[100];
+    jeBinInfo *binfo;
+
+    size_t len = sizeof(jemalloc_quantum);
+    je_mallctl("arenas.quantum", &jemalloc_quantum, &len, NULL, 0);
+    // lg-quantum can be 3 or 4
+    assert((jemalloc_quantum == 8) || (jemalloc_quantum == 16));
+
+    unsigned nbins;
+    sz = sizeof(nbins);
+    assert(!je_mallctl("arenas.nbins", &nbins, &sz, NULL, 0));
+    arena_bin_conf.bin_info = zcalloc(sizeof(jeBinInfo) * nbins);
+    for (unsigned j = 0; j < nbins; j++) {
+        binfo = &arena_bin_conf.bin_info[j];
+        /* The size of the current bin */
+        snprintf(buf, sizeof(buf), "arenas.bin.%d.size", j);
+        sz = sizeof(size_t);
+        assert(!je_mallctl(buf, &binfo->reg_size, &sz, NULL, 0));
+
+        /* Number of regions per slab */
+        snprintf(buf, sizeof(buf), "arenas.bin.%d.nregs", j);
+        sz = sizeof(uint32_t);
+        assert(!je_mallctl(buf, &binfo->nregs, &sz, NULL, 0));
+        binfo->len = binfo->reg_size * binfo->nregs;
+        /* Mib of fetch number of used regions in the bin */
+        snprintf(buf, sizeof(buf), "stats.arenas." STRINGIFY(ARENA_TO_QUERY) ".bins.%d.curregs", j);
+        sz = sizeof(size_t);
+        binfo->info_keys.curr_regs_key_len = sizeof(binfo->info_keys.curr_regs_key) / sizeof(size_t);
+        assert(!je_mallctlnametomib(buf, binfo->info_keys.curr_regs_key, &binfo->info_keys.curr_regs_key_len));
+        /* Mib of fetch number of current slabs in the bin */
+        snprintf(buf, sizeof(buf), "stats.arenas." STRINGIFY(ARENA_TO_QUERY) ".bins.%d.curslabs", j);
+        binfo->info_keys.curr_slabs_key_len = sizeof(binfo->info_keys.curr_slabs_key) / sizeof(size_t);
+        assert(!je_mallctlnametomib(buf, binfo->info_keys.curr_slabs_key, &binfo->info_keys.curr_slabs_key_len));
+        /* Mib of fetch nonfull slabs */
+        snprintf(buf, sizeof(buf), "stats.arenas." STRINGIFY(ARENA_TO_QUERY) ".bins.%d.nonfull_slabs", j);
+        binfo->info_keys.nonfull_slabs_key_len = sizeof(binfo->info_keys.nonfull_slabs_key) / sizeof(size_t);
+        assert(!je_mallctlnametomib(buf, binfo->info_keys.nonfull_slabs_key, &binfo->info_keys.nonfull_slabs_key_len));
+
+        /* Mib of fetch num of alloc op */
+        snprintf(buf, sizeof(buf), "stats.arenas." STRINGIFY(ARENA_TO_QUERY) ".bins.%d.nmalloc", j);
+        binfo->info_keys.nmalloc_key_len = sizeof(binfo->info_keys.nmalloc_key) / sizeof(size_t);
+        assert(!je_mallctlnametomib(buf, binfo->info_keys.nmalloc_key, &binfo->info_keys.nmalloc_key_len));
+        /* Mib of fetch num of dealloc op */
+        snprintf(buf, sizeof(buf), "stats.arenas." STRINGIFY(ARENA_TO_QUERY) ".bins.%d.ndalloc", j);
+        binfo->info_keys.ndealloc_key_len = sizeof(binfo->info_keys.ndealloc_key) / sizeof(size_t);
+        assert(!je_mallctlnametomib(buf, binfo->info_keys.ndealloc_key, &binfo->info_keys.ndealloc_key_len));
+        // verify the reverse map of reg_size to bin index
```

Comment:
The code styling as of right now always uses C style comments.

```suggestion
        /* verify the reverse map of reg_size to bin index */
```